### PR TITLE
Automatically promote releases after approving issue

### LINKED ISSUE #2122: Refactor data prepper release process
**Is your feature request related to a problem? Please describe.**
There has been a development in the way artifacts are released. We would like to re-do the data-prepper release process to use the new process https://github.com/opensearch-project/opensearch-build/issues/1234

Old release process jenkins file : https://github.com/opensearch-project/opensearch-build/blob/main/jenkins/data-prepper/release-data-prepper-all-artifacts.jenkinsfile

**Describe the solution you'd like**
1. Add release-drafter GitHub action workflow that will build the artifacts and upload the artifacts to a draft release. [Sample workflow](https://github.com/opensearch-project/spring-data-opensearch/blob/main/.github/workflows/release-drafter.yml)
2. Add a new jenkins file to this repository that downloads those artifacts from the above `draft release` and uses a new lib called `publishToMaven` that autopublishes the maven artifacts to maven. Similary, other libraries such as copyContainer for copying container images to different platforms. This jenkins File uses draft release as a trigger. 

In this new process, all that a maintainer needs to do is push a tag and the releasing artifacts to different platforms will be taken care by the GHA and jenkins workflows. 

**Describe alternatives you've considered (Optional)**
Continue to use what we have.

**Additional context**

## Design

See comment: https://github.com/opensearch-project/data-prepper/issues/2122#issuecomment-1662414428

## Tasks

- [ ] Update GitHub Action for releasing to include parameters, new approval workflow, and creating the draft release.
- [ ] https://github.com/opensearch-project/opensearch-build/issues/3842
- [ ] Add Jenkins token to the GitHub secrets for the Data Prepper repository
- [ ] Add webhook to Data Prepper GitHub project on either tagging or drafting of releases
- [ ] #3108

### PR DESCRIPTION
**Pull request by dlvenable**
### Description

Automatically promote releases by creating a draft release of the build with release metadata. This will trigger a Jenkins job on https://build.ci.opensearch.org/ to promote. Performing this build will require the approval of two maintainers of the project.
 
### Issues Resolved

Resolves #2122.
 
### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has a documentation issue. Please link to it in this PR.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: .github/workflows/release.yml
Comment (asifsmohammed) [inline .github/workflows/release.yml:144]: Steps from here will only be executed after getting 2 approvals right?

File: .github/workflows/release.yml
Comment (asifsmohammed) [inline .github/workflows/release.yml:6]: How do we provide this input?

File: .github/workflows/release.yml
Comment (dlvenable) [reply .github/workflows/release.yml:144]: Yes, this will require 2 approvals. The workflow initiator may be one of those approvers as well.

File: .github/workflows/release.yml
Comment (dlvenable) [reply .github/workflows/release.yml:6]: Right now, we run the release build using the GitHub UI. With this change, GitHub will provide an option to specify this value. Unfortunately, I don't think GitHub Actions support "boolean" types. So you have to type in "true" or "false".

File: .github/workflows/release.yml
Comment (dlvenable) [inline .github/workflows/release.yml:109]: Our smoke tests currently fail the build. You can see in the text whether it passed or not, but something is causing them to fail. So for now, the promote job will run whether the smoke tests report failure or success.

**Review by asifsmohammed — APPROVED** [review]

**Review by chenqi0805 — APPROVED** [review]

**Pull request merged**
